src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java

```diff
@@ -915,6 +916,73 @@ public void testContainerWithHostConfig() throws Exception {
   }
 
   @Test
+  public void testEventStream() throws Exception {
+    sut.pull("busybox");
+    EventStream eventStream = sut.events();
+    final ContainerConfig config = ContainerConfig.builder()
+            .image("busybox")
+            .build();
+    final ContainerCreation container = sut.createContainer(config, randomName());
+    sut.startContainer(container.id());
+
+    final Event createEvent = eventStream.next();
+    assertThat(createEvent.status(), equalTo("create"));
+    assertThat(createEvent.id(), equalTo(container.id()));
+    assertThat(createEvent.from(), startsWith("busybox:"));
+    assertThat(createEvent.time(), notNullValue());
+
+    final Event startEvent = eventStream.next();
+    assertThat(startEvent.status(), equalTo("start"));
+    assertThat(startEvent.id(), equalTo(container.id()));
+    assertThat(startEvent.from(), startsWith("busybox:"));
+    assertThat(startEvent.time(), notNullValue());
+
+    eventStream.close();
+  }
+
+  @Test
+  public void testEventStreamWithSinceTime() throws Exception {
+    Thread.sleep(1000); // ensure we push to the next second
+                        // so we don't get events from the last test
+    final Date date = new Date();
+    sut.pull("busybox");
+    final ContainerConfig config = ContainerConfig.builder()
+            .image("busybox")
+            .build();
+    final ContainerCreation container = sut.createContainer(config, randomName());
+    sut.startContainer(container.id());
+
+    EventStream eventStream = sut.events(DockerClient.EventsParam.since(date.getTime() / 1000));
+
+    final Event pullEvent = eventStream.next();
+    assertThat(pullEvent.status(), equalTo("pull"));
+    assertThat(pullEvent.id(), equalTo("busybox"));
+
+    final Event createEvent = eventStream.next();
+    assertThat(createEvent.status(), equalTo("create"));
+    assertThat(createEvent.id(), equalTo(container.id()));
+    assertThat(createEvent.from(), startsWith("busybox:"));
+    assertThat(createEvent.time(), notNullValue());
+
+    final Event startEvent = eventStream.next();
+    assertThat(startEvent.status(), equalTo("start"));
+    assertThat(startEvent.id(), equalTo(container.id()));
+    assertThat(startEvent.from(), startsWith("busybox:"));
+    assertThat(startEvent.time(), notNullValue());
+
+    eventStream.close();
+  }
+
+  @Test(timeout = 5000)
+  public void testEventStreamWithUntilTime() throws Exception {
+    EventStream eventStream = sut.events(DockerClient.EventsParam.until((new Date().getTime() + 2000) / 1000));
+
+    while (eventStream.hasNext()) {
+      eventStream.next();
+    }
+  }
```

Comment:
Since it isn't 100% clear what this test does, I'll explain here:

Basically, the Docker events API provides a way to end the streaming at a given timestamp. This method calls the streaming events API with a timestamp 2 seconds from now, but sets the test to timeout in 5 seconds. Basically, if the client doesn't pass the parameter to the API, the test will fail with a timeout (because it'll still be waiting for events to stream after 5 seconds). But if all is well, the Docker daemon should close the stream in ~2 seconds.